### DISCUSSION TITLE
Change the shape of highlight annotation

### DISCUSSION THREAD

**Discussion by dawn-fm** [sumatrapdfreader/sumatrapdf#4386] [Q&A] [answered]
Hi all,

I wonder if I can customize the shape of highlight annotation.

As you surely know, Figure 1 below is the original highlight shape Sumatrapdf provides, which is a bit round and not compact to the text,

![image](https://github.com/user-attachments/assets/3c73bd41-399f-4a4b-b458-c8d283cd4c3c)
(Figure 1)

and I would like to change the shape to be **squared and compact** as in Figure 2 (actually this is highlighted by pdf expert in _macos_)

![image](https://github.com/user-attachments/assets/5c7bcc7d-9789-484a-99a0-90f3765bea73)
(Figure 2)

Does anyone know the solutions to do this?

Thanks,

**Comment by GitHubRulesOK** [comment]
The highlighter is based on MuPDF, and has over time changed shape and height in SumatraPDF, so the rounding off in older yellow was less noticeable than current version orange which I thought was proportional to the area coverage !! However if a chord it will be the horizonal offset makes a difference to the end curves, and the horizontal sides are not the same.

![image](https://github.com/user-attachments/assets/3131b0a8-7602-40f0-9aaa-18cb70f55770)

If I change the old colour you can see the difference so it looks like the shape has been improved !

![image](https://github.com/user-attachments/assets/39b223c0-105b-4d40-a58a-f8401fa775ac)

**Reply by dawn-fm** [reply]
@GitHubRulesOK Thnx bro, then can I change the highlight shape? What I want in the end is changing the shape to be squared.

**Reply by GitHubRulesOK** [reply]
@kjk
Legacy SMK was simple rectangles and I think it was the shift to MuPDF has the rounded formats (currently can not test MuPDF-GL to confirm it is their default shape) is it easy to over-ride for a user choice of flat or rounded? I think there was already an open issue. 


Ah yes it was vertical https://github.com/sumatrapdfreader/sumatrapdf/issues/2489

**Reply by dawn-fm** [reply]
@GitHubRulesOK okay, then are there any chances to make such features?

**Reply by GitHubRulesOK** [reply, marked as answer]
@dawn-fm 
Ok I fired up MuPDF (the core PDF annotation editor used as part of SumatraPDF and as suspected that is where the shape is defined.
Comparing SumatraPDF with MuPDF
![image](https://github.com/user-attachments/assets/bf839ba4-57c7-4c71-b521-2cb37106d5ba)
![image](https://github.com/user-attachments/assets/d7056546-8950-4fc5-bd45-59d0a29a95bc)

Thus would need unknown adjustments in SumatraPDF to do it different, the developer @kjk would need to estimate the chances of change. Any one can provide the code as a pull request but needs to test across the many odd usages such as vertical, rotated, diagonal  and same for areas etc.

**Reply by dawn-fm** [reply]
@GitHubRulesOK okay thank you for your response!
